### DISCUSSION TITLE
add DiscountCustomerGetsInput field to DiscountCustomerGetsInput

### DISCUSSION THREAD

**Discussion by silentmado** [Shopify/function-examples#451] [Feedback: Discounts APIs]
Why DiscountCodeAppInput doesn't contain a field DiscountCustomerGetsInput to select either a discount should be applied on OneTimePurchase, Subscription or Both ?

**Comment by jharlap** [comment]
Your Discount Function returns the discount a customer gets as part of the `discounts` field in [FunctionRunResult](https://shopify.dev/docs/api/functions/reference/product-discounts/graphql/functionrunresult). If you want to know if a cart line item is a one time purchase or a subscription, you can examine the `sellingPlanAllocation` property of each cart line.

**Reply by bkspace** [reply]
This does not work because you can only target product variant ids and not specific cart line ids - it's currently not possible to target based on sellingPlan unfortunately!

**Comment by jharlap** [comment]
It is now possible to use cart line targets in Product Discount Functions.
